vispy/gloo/glir.py

```diff
@@ -876,18 +905,22 @@ def set_interpolation(self, min, mag):
 class GlirTexture2D(GlirTexture):
     _target = gl.GL_TEXTURE_2D
     
-    def set_size(self, shape, format):
+    def set_size(self, shape, format, internalformat):
         # Shape is height, width
         format = as_enum(format)
-        if (shape, format) != self._shape_format:
-            self._shape_format = shape, format
+        if internalformat is not None:
+            internalformat = as_enum(internalformat)
+        else:
+            internalformat = format
```

Comment:
`internalformat = format if internalformat is None else as_enum(internalformat)`